network/wsNetwork.go

```diff
@@ -470,6 +472,13 @@ type WebsocketNetwork struct {
 
 	// resolveSRVRecords is a function that resolves SRV records for a given service, protocol and name
 	resolveSRVRecords func(service string, protocol string, name string, fallbackDNSResolverAddress string, secure bool) (addrs []string, err error)
+
```

Comment:
I think this is too much complication in the network layer.
I have a prototype suggestion here: 
https://github.com/iansuvak/go-algorand/pull/4

Keep the accounting similar to https://github.com/algorand/go-algorand/pull/5428 and use a callback as in https://github.com/iansuvak/go-algorand/pull/4